### ISSUE TITLE
Standardize params names across methods, problems etc.

### ISSUE THREAD

**Issue by sesevgen** [IBM/mi-prometheus#73]
There is inconsistent naming across params. For example, directories are sometimes called 'dir' and sometimes called 'folder'. We should perhaps decide on a unified set of standard names for source and target folders etc, and then change __init__ and configs to reflect it.

**Comment by tkornut** [comment]
I totally agree, I already started that with CLEVR, Sort-of-CLEVR, MNIST and CIFAR10.

I used "data_folder", with default value set to "~/data/mnist" (in the case of MNIST;) )

Besides, along with the introduction of helpers that can operate directly on problems/models, we have made a decision that all problems/models should provide default values for all parameters, enabling to instantiate them without configuration file.

Please refer to:
https://github.com/IBM/mi-prometheus/blob/194cedec3431c7b56d1c1b2fed1a6e75e9fec265/miprometheus/problems/image_to_class/mnist.py#L91

@sesevgen Please keep that in mind now, probably I should create a separate issues for this...

**Comment by vmarois** [comment]
Agree on this as well. 
Alongside unifying the names used for the 'common' arguments, we should also make more explicit the several arguments that a Problem / Model class expects. This is indicated in #34 🙂
